schema: add scylla specific options to schema description

Jadw1 opened the pull request:
Add `paxos_grace_seconds`, `tombstone_gc`, `cdc` and `synchronous_updates` options to schema description.

Fixes: #12389
Fixes: https://github.com/scylladb/scylla-enterprise/issues/2979

Jadw1 commented:
@bhalevy let me know if I missed any option

scylladb-promoter commented:
CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/1895/

bhalevy commented on schema/schema.cc:
```diff
@@ -906,8 +917,15 @@ std::ostream& schema::describe(replica::database& db, std::ostream& os, bool wit
     os << "\n    AND read_repair_chance = " << read_repair_chance();
-    os << "\n    AND speculative_retry = '" << speculative_retry().to_sstring() << "';";
-    os << "\n";
+    os << "\n    AND speculative_retry = '" << speculative_retry().to_sstring() << "'";
```
> Add `speculative_retry`, `paxos_grace_seconds`, `cdc` and `synchronous_updates`
options to schema description.

It looks like `speculative_retry` was already described.

bhalevy commented:
> @bhalevy let me know if I missed any option

Based on https://github.com/scylladb/scylladb/blob/master/cql3/statements/cf_prop_defs.cc:
`paxos_grace_seconds`

Also, please also examine `operator<<(std::ostream& os, const schema& s)` to see if it prints the same information as `schema::describe()`.

Jadw1 commented:
> Based on https://github.com/scylladb/scylladb/blob/master/cql3/statements/cf_prop_defs.cc: `paxos_grace_seconds`

`paxos_grace_seconds` is present in a describe.
 
> Also, please also examine `operator<<(std::ostream& os, const schema& s)` to see if it prints the same information as `schema::describe()`.

`schema::describe()` doesn't print: `tombstoneGcOptions`, `keyValidator`, `minCompactionThreshold`, `maxCompactionThreshold` and it prints `cdc` only if `cdc_options().enabled()`

`operator<<(...)` doesn't print: `paxos_grace_seconds`, `crc_check_chance`


I've updated the test to include added options, but the CI won't pass because there is sth wrong with boost test's cql env.
1. Creating table with specified `paxos_grace_seconds` doesn't take any effect. The value of `std::optional<int32_t> schema::raw_schema::_paxos_grace_seconds` is `std::nullopt_t`
2. The test cannot read `speculative_retry` because `db::find_tag()` segfaults. The entry from schema's extensions is found but casting `schema_extension` to `tags_extension` fails. The function uses `static_pointer_cast` (thus the segfault) and when I've changed it to dynamic cast the result pointer is null. But the original pointer from extensions map is not null.

scylladb-promoter commented:
CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2006/

Jadw1 commented:
Fixed cql-test env by simply adding `db::extensions` to `cql_test_config`.

Added `tombstone_gc`: https://github.com/scylladb/scylla-enterprise/issues/2979

scylladb-promoter commented:
CI state **`FAILURE`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2206/

scylladb-promoter commented:
CI state **`SUCCESS`** - https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/2211/

Jadw1 commented:
@bhalevy can we merge this?

bhalevy commented:
> @bhalevy can we merge this?

It's not up to me...

@scylladb/scylla-maint please merge
